pkgs/development/python-modules/pyp/default.nix

```
@@ -1,56 +1,68 @@
 { lib
+, astunparse
+, bc
 , buildPythonPackage
 , fetchFromGitHub
-, pytestCheckHook
-, coreutils
-, pythonOlder
-, astunparse
 , flit-core
 , jq
-, bc
+, pytestCheckHook
+, pythonOlder
 }:
 
-buildPythonPackage rec {
-  pname = "pyp";
-  version = "1.2.0";
-  format = "pyproject";
+let
+  finalAttrs = {
+    pname = "pyp";
+    version = "1.2.0";
 
-  disabled = pythonOlder "3.6";
+    src = fetchFromGitHub {
+      owner = "hauntsaninja";
+      repo = "pyp";
+      rev = "refs/tags/v${finalAttrs.version}";
+      hash = "sha256-hnEgqWOIVj2ugOhd2aS9IulfkVnrlkhwOtrgH4qQqO8=";
+    };
 
-  src = fetchFromGitHub {
-    owner = "hauntsaninja";
-    repo = pname;
-    rev = "refs/tags/v${version}";
-    hash = "sha256-hnEgqWOIVj2ugOhd2aS9IulfkVnrlkhwOtrgH4qQqO8=";
-  };
+    pyproject = true;
 
-  nativeBuildInputs = [
-    flit-core
-  ];
-
-  propagatedBuildInputs = lib.optionals (pythonOlder "3.9") [
-    astunparse
-  ];
-
-  preCheck = ''
-    export PATH=$out/bin:$PATH
-  '';
-
-  nativeCheckInputs = [
-    pytestCheckHook
-    coreutils
-    jq
-    bc
-  ];
-
-  pythonImportsCheck = [
-    "pyp"
-  ];
-
-  meta = with lib; {
-    description = "Easily run Python at the shell! Magical, but never mysterious";
-    homepage = "https://github.com/hauntsaninja/pyp";
-    license = licenses.mit;
-    maintainers = with maintainers; [ rmcgibbo ];
-   };
-}
+    disabled = pythonOlder "3.6";
+
+    nativeBuildInputs = [
```

```suggestion
    build-system = [
```
We have changed to use `build-system` instead of `nativeBuildInputs` recently.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

---

O.O